jax/numpy/lax_numpy.py

```diff
@@ -2541,17 +2541,18 @@ def diag(v, k=0):
     raise ValueError("diag input must be 1d or 2d")
 
 
+@jit
+def _polyval(p, x):
+  shape = lax.broadcast_shapes(p.shape[1:], x.shape)
+  dtype = result_type(p, x)
+  y = lax.full_like(x, 0, shape=shape, dtype=dtype)
+  y, _ = lax.scan(lambda y, p: (y * x + p, None), y, p)
```

Comment:
I suspect the downside here is that this will be slower on GPU because loops are often slow on GPU. Perhaps that's an acceptable tradeoff? Or perhaps there is something to fixing lax.scan to unroll a little bit internally on GPU?

Reply:
Unrolling a bit on GPU sounds reasonable to me! (Though I wonder if that's also an easy XLA:GPU-level change, for cases when the While HLO in question is really a "for loop with static trip count", as is already detected in some optimizations.)

Reply:
I thought XLA:GPU could handle static loop sizes well, but ouch, it is slow!

256 ms to do a single iteration on GPU, vs 31 us on CPU!